### PR TITLE
Schemas + Handler

### LINKED ISSUE #216: Handler rework: revoke dynamic setup
# Feature
## Desired Behavior / Functionality
The handler was reworked to allow for a dynamic setup to amount for different attacks. This made it complicated beyond what was necessary.

## What Needs to Be Done
Make an abstract handler and have individual handlers for the different attacks.

## How Can It Be Tested
Examples should run.

### LINKED ISSUE #219: Implement schemas for data structures
# Feature
## Desired Behavior / Functionality

It is very hard for a new user to understand what is required as input to LeakPro. This will also ensure compliance.

## What Needs to Be Done
Introduce schemas for all data structures. 

## How Can It Be Tested
Everything should still run.

### LINKED ISSUE #228: single shadow model error
# Issue
## Problem Description
When using a single shadow model in online RMIA and LiRA, it should break gracefully!

### PR DESCRIPTION
**Pull request by johanos1**
# Description

Summary of changes

* The attack handler for MIA is no longer dynamically assigned.
* Schemas have been introduced that define the data structures that goes into storage and is exchanged between different modules. 
* Run-time checks for compatibility between config params have been introduced.
* Introducing attack schemas for all MIA attacks with proper checks. This will pave the road towards Optuna integration.
* Create the image modality extension (issue #152). This will be finalized in another PR 

## Resolved Issues

- [ ] fixes #219 
- [ ] fixes #216 
- [ ] fixes #228 

## How Has This Been Tested?
- All MIA examples are running.
- Unit tests are passing.

### CODE REVIEW

File: leakpro/input_handler/modality_extensions/image_extension.py
Comment (fazelehh) [inline leakpro/input_handler/modality_extensions/image_extension.py:1]: This is not relevant to this PR.

File: leakpro/input_handler/modality_extensions/modality_extension.py
Comment (fazelehh) [inline leakpro/input_handler/modality_extensions/modality_extension.py:1]: Not relevant to this PR

File: leakpro/input_handler/modality_extensions/tabular_extension.py [hunk 1]
```diff
@@ -36,6 +41,21 @@ def __init__(self:Self) -> None:
         else:
             raise ValueError("Data object must contain dec_to_onehot dict.")
 
+    def augmentation(self:Self, data:Tensor, n_aug:int) -> Tensor:
```
Comment (fazelehh) [inline leakpro/input_handler/modality_extensions/tabular_extension.py:44]: Not a relevant function to this PR

**Review by fazelehh — CHANGES_REQUESTED** [review]
Changes regarding Image Extension should not be included in this PR.

File: leakpro/input_handler/modality_extensions/image_extension.py
Comment (johanos1) [reply leakpro/input_handler/modality_extensions/image_extension.py:1]: I added some text to the PR description.

File: leakpro/input_handler/modality_extensions/tabular_extension.py (hunk 1 above)
Comment (johanos1) [reply leakpro/input_handler/modality_extensions/tabular_extension.py:44]: Same as above.

**Review by fazelehh — COMMENTED** [review]
It looks like audit.yaml is fully deleted, not just removed from the PR

File: examples/mia/cifar/cifar_handler.py [hunk 2]
```diff
@@ -46,22 +46,26 @@ def train(
...
                 optimizer.step()
 
                 # Accumulate performance of shadow model
                 train_acc += pred.eq(labels.data.view_as(pred)).sum()
+                total_samples += labels.size(0)
                 train_loss += loss.item()
-
+                
+        avg_train_loss = train_loss / total_samples
+        train_accuracy = train_acc / total_samples
```
Comment (henrikfo) [inline examples/mia/cifar/cifar_handler.py:67]: Was going to make an issue regarding this, that acc and loss was not scaled.
But this is should give the same result as in "cifar_model_preparation.py" for accuracy, but not loss right?

**Review by henrikfo — APPROVED** [review]
Looks good!

**Review by fazelehh — CHANGES_REQUESTED** [review]
One error in the HSJ file.
Some minor comments.
And one question: Right now, users have to check each attack individually to see which variables can be modified. Is there an issue addressing this problem?

**Review by fazelehh — APPROVED** [review]
